packages/kbn-esql-ast/src/types.ts

```diff
@@ -59,6 +61,16 @@ export interface ESQLFunction extends ESQLAstBaseItem {
   args: ESQLAstItem[];
 }
 
+export interface ESQLInlineCast<ValueType = ESQLAstItem> extends ESQLAstBaseItem {
+  type: 'inlineCast';
+  value: ValueType;
+  castType: string;
+}
+
+export interface ESQLUnknownItem extends ESQLAstBaseItem {
+  type: 'unknown';
```

Comment:
Why do we need this type? Can you give me examples of `unknown type`

Reply:
You can think of it like an escape hatch. for the AST generator code if it doesn't recognize something in the parse tree. This could happen if the grammar changes and the AST code isn't updated or if ANTLR has an idiosyncrasy (which has happened before).

Previously, we used `undefined` to handle these cases. I like having a new AST node type better because
- it encodes information about what surprised the AST generator
- it may alert us to changes we need to make to the AST generator if we see it in an AST
- it removes `undefined` from the AST typings

As discussed in Slack, I will add a comment to the code so future devs can follow the thinking!

Reply:
Thanx for the comment (both in the gh and the code, makes sense and now is also easier to understand in the code. Thank you!